storage/src/path/buf.rs

```diff
@@ -128,3 +128,68 @@ impl<'a> IntoSplitPath for &'a PartialPath<'_> {
         self
     }
 }
+
+/// A RAII guard that resets a path buffer to its original length when dropped.
```

Comment:
Actually, this isn't completely true. If you create a PathGuard and then shorten PathBuf, it will remain shortened after dropping it, as the truncate will have no effect.